src/randgraphs.jl

```diff
@@ -107,9 +107,13 @@ function _suitable(edges::Set{Edge}, potential_edges::Dict{Int, Int})
     return false
 end
 
-function _try_creation(n::Int, k::Int)
+function _try_creation(n::Int,@compat k::Union{Int,Array{Int}})
     edges = Set{Edge}()
-    stubs = repmat([1:n;], k)
+    if typeof(k) == Int
+        stubs = repmat([1:n;], k)
+    else
+        stubs = vcat([fill(i, k[i]) for i=1:n]...)
```

Comment:
If I'm reading this right, this creates a dense array of `nxn`? That means it won't work very well with large graphs.